src/core/bittorrent/torrenthandle.cpp

```diff
@@ -1560,6 +1562,23 @@ void TorrentHandle::handleFileRenamedAlert(libtorrent::file_renamed_alert *p)
         // Renaming a file corresponds to changing the save path
         m_session->handleTorrentSavePathChanged(this);
     }
+
+    --m_renameCount;
+    if (!isMoveInProgress() && m_renameCount == 0) {
+        while (!m_torrentFinishedTriggers.isEmpty())
+            m_torrentFinishedTriggers.takeFirst()();
+    }
+}
+
+void TorrentHandle::handleFileRenameFailedAlert(libtorrent::file_rename_failed_alert *p)
+{
+    Q_UNUSED(p);
+
+    --m_renameCount;
+    if (!isMoveInProgress() && m_renameCount == 0) {
+        while (!m_torrentFinishedTriggers.isEmpty())
+            m_torrentFinishedTriggers.takeFirst()();
```

Comment:
Hell, I found the problem here!
A trigger can start moving/renaming and it will corrupt the following triggers. To avoid this, we must do so:
```c++
while (!isMoveInProgress() && (m_renameCount == 0) && !m_torrentFinishedTriggers.isEmpty())
    m_torrentFinishedTriggers.takeFirst()();
```